alphadia/workflow/peptidecentric.py

```diff
@@ -537,7 +553,53 @@ def first_recalibration_and_optimization(
 
         for optimizers in ordered_optimizers:
             for optimizer in optimizers:
-                optimizer.step(precursor_df_filtered, fragments_df_filtered)
+                optimizer.step(
+                    precursor_df_filtered, fragments_df_filtered, record_step=False
+                )
+
+        precursor_df_filtered["precursor_idx"] = precursor_df_filtered[
+            "precursor_idx"
+        ].map(self.randomized_idx_mapping)
+        try:
+            # Takes the target number of precursors if at least the target is found
+            self.precursor_cutoff_idx = (
+                precursor_df_filtered.sort_values(by="precursor_idx")
+                .iloc[self.config["calibration"]["optimization_lock_target"]]
+                .precursor_idx
+            )
+        except IndexError:
+            # Otherwise takes all of them
+            self.precursor_cutoff_idx = (
+                precursor_df_filtered.sort_values(by="precursor_idx")
+                .iloc[-1]
+                .precursor_idx
+                + 1
+            )
+
+        return precursor_df_filtered, fragments_df_filtered
+
+    def set_reduced_optimization_lock(self):
+        self.reduced_optimization_lock_library_precursor_df = (
+            self.optimization_lock_library_precursor_df[
+                self.optimization_lock_library_precursor_df["precursor_idx"]
+                < self.precursor_cutoff_idx
+            ]
+        )
+
+        fragment_cutoff_idx = self.reduced_optimization_lock_library_precursor_df[
```

Comment:
this is a reeeeally long variable name ..

think about introucing `optlock` as subsitute for `optimization_lock` throughout all the codebase (if you choose to do so, please do the renaming in a dedicated PR)

then this could become `red_optlock_lib_precursor_df`